src/argilla/server/contexts/datasets.py

```diff
@@ -536,24 +535,126 @@ async def create_records(
     await db.commit()
 
 
+async def _exists_records_with_ids(db: "AsyncSession", dataset_id: UUID, records_ids: List[UUID]) -> List[UUID]:
+    result = await db.execute(select(Record.id).filter(Record.dataset_id == dataset_id, Record.id.in_(records_ids)))
+    return result.scalars().all()
+
+
+async def update_records(
+    db: "AsyncSession", search_engine: "SearchEngine", dataset: Dataset, records_update: "RecordsUpdate"
+) -> None:
+    records_ids = [record_update.id for record_update in records_update.items]
+
+    if len(records_ids) != len(set(records_ids)):
+        raise ValueError("Found duplicate records IDs")
+
+    existing_records_ids = await _exists_records_with_ids(db, dataset_id=dataset.id, records_ids=records_ids)
+    non_existing_records_ids = set(records_ids) - set(existing_records_ids)
+
+    if len(non_existing_records_ids) > 0:
+        raise ValueError("Found records that do not exist")
+
+    # Lists to store the records that will be updated in the database or in the search engine
+    records_update_objects: List[Dict[str, Any]] = []
+    records_search_engine_update: List[UUID] = []
+    records_delete_suggestions: List[UUID] = []
+
+    # Cache dictionaries to avoid querying the database multiple times
+    metadata_properties: Dict[str, Union[MetadataProperty, Literal["extra"]]] = {}
+    questions: Dict[UUID, Question] = {}
+
+    suggestions = []
+    for record_i, record_update in enumerate(records_update.items):
+        params = record_update.dict(exclude_unset=True)
+
+        if "metadata_" in params and (metadata := params["metadata_"]) is not None:
+            try:
+                metadata_properties = await _validate_metadata(db, dataset, metadata, metadata_properties)
+            except ValueError as err:
+                raise ValueError(f"Provided metadata for record at position {record_i} is not valid: {err}") from err
+            records_search_engine_update.append(record_update.id)
+
+        if record_update.suggestions is not None:
+            params.pop("suggestions")
+            for suggestion_i, suggestion in enumerate(record_update.suggestions):
+                try:
+                    questions = await _validate_suggestion(db, suggestion, questions)
+                    suggestions.append(Suggestion(record_id=record_update.id, **suggestion.dict()))
+                except ValueError as err:
+                    raise ValueError(
+                        f"Provided suggestion for record at position {record_i} and suggestion at position "
+                        f"{suggestion_i} is not valid: {err}"
+                    ) from err
```

Comment:
I'm going to add a check for checking that there is just one suggestion for each question_id per record, otherwise we can get an `IntegrityError` later